src/main/java/edu/ksu/canvas/attendance/controller/CourseConfigurationController.java

```diff
@@ -93,6 +116,44 @@ public ModelAndView synchronizeWithCanvas(@PathVariable String sectionId) throws
         return page;
     }
 
+    @RequestMapping(value = "/{sectionId}/save", params = "pushGradesToCanvas", method = RequestMethod.POST)
+    public ModelAndView pushGradesToCanvas(@PathVariable String sectionId, @ModelAttribute("courseConfigurationForm") @Valid CourseConfigurationForm classSetupForm, BindingResult bindingResult) throws NoLtiSessionException{
```

Comment:
I think sectionId can be a long in the method signature. Spring should set it appropriately automatically. This will avoid parsing multiple times in the body